### DISCUSSION TITLE
How to using environment variables on Github actions? Unrecognized named-value: 'GITHUB_REPOSITORY'.

### DISCUSSION THREAD

**Discussion by tomac4t** [community/community#26511] [Actions]
<p>I read the Github Help document <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables" rel="nofollow noopener">“using environment variables”</a> and add “env” on the yaml file, becuase I want to upload the assets by using REST API. But I got a error (Link: <a href="https://github.com/tomac4t/srain/actions/runs/79918460" rel="nofollow noopener">https://github.com/tomac4t/srain/actions/runs/79918460</a>). The error said, “The workflow is not valid. .github/workflows/release.yml (Line: 9, Col: 16): Unrecognized named-value: ‘GITHUB_REPOSITORY’. Located at position 1 within expression: GITHUB_REPOSITORY”.</p>
<p>I check the doucment and these is not typo about these environment variables. I don’t know how to use <span class="math">{{ GITHUB\_REPOSITORY }} on Github actions. (Looks&amp;nbsp;</span>{{ secrets.GITHUB_TOKEN }} work.)</p>
<p>Screenshot:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/df3bd102a21f22ee2df4439c8ef0c7efd073d9bc.png" data-download-href="/uploads/short-url/vQODbCxqw3rUiaxuvSxKEUD8Xj6.png?dl=1" title="Screenshot_2020-04-16 tomac4t srain.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/47532693/181098104-85111177-5a4c-4fd4-ba64-790f402604ae.png" alt="Screenshot_2020-04-16 tomac4t srain.png" data-base62-sha1="vQODbCxqw3rUiaxuvSxKEUD8Xj6" width="690" height="396" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/df3bd102a21f22ee2df4439c8ef0c7efd073d9bc_2_690x396.png, https://aws1.discourse-cdn.com/github/original/2X/d/df3bd102a21f22ee2df4439c8ef0c7efd073d9bc.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/d/df3bd102a21f22ee2df4439c8ef0c7efd073d9bc.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/df3bd102a21f22ee2df4439c8ef0c7efd073d9bc_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_2020-04-16 tomac4t srain.png</span><span class="informations">965×555 45.2 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment, marked as answer]
<p>You’re not in a shell (or similar) context, so you probably have to use the “env” context like so: <span>${{ env.GITHUB_REPOSITORY }}</span></p>
<p>See: <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#env-context" rel="nofollow noopener">https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#env-context</a></p>

**Comment by tomac4t** [comment]
<p>Thanks! I found $GITHUB_REPOSITORY just fine in not the shell context.</p>
<p>Using Github context <span class="math">{{ github.repository }} is fine, instead of&amp;nbsp; </span>{{ GITHUB_REPOSITORY }}.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/8/88b295f88241c93991ab03ef761e1bc104150b6e.png" data-download-href="/uploads/short-url/jvhymTncGdSoBia9U3iNWq82X1k.png?dl=1" title="Screenshot_2020-04-17 tomac4t srain.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/47532693/181098107-a200d89f-5ece-4686-a156-26a92ed412da.png" alt="Screenshot_2020-04-17 tomac4t srain.png" data-base62-sha1="jvhymTncGdSoBia9U3iNWq82X1k" width="690" height="326" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/8/88b295f88241c93991ab03ef761e1bc104150b6e_2_690x326.png, https://aws1.discourse-cdn.com/github/optimized/2X/8/88b295f88241c93991ab03ef761e1bc104150b6e_2_1035x489.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/8/88b295f88241c93991ab03ef761e1bc104150b6e.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/8/88b295f88241c93991ab03ef761e1bc104150b6e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot_2020-04-17 tomac4t srain.png</span><span class="informations">1340×635 50.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
